clang/lib/3C/ConstraintResolver.cpp

```diff
@@ -185,13 +185,20 @@ CVarSet ConstraintResolver::getInvalidCastPVCons(CastExpr *E) {
   return {P};
 }
 
-// Returns a set of ConstraintVariables which represent the result of
-// evaluating the expression E. Will explore E recursively, but will
-// ignore parts of it that do not contribute to the final result
-CVarSet ConstraintResolver::getExprConstraintVars(Expr *E) {
-  CVarSet EmptyCSet;
-  auto &ABInfo = Info.getABoundsInfo();
+inline CSetBkeyPair convertToCSetBKeyPair(const CVarSet &Vars) {
+  BKeySet EmptyBSet;
+  EmptyBSet.clear();
+  return std::make_pair(Vars, EmptyBSet);
+}
 
+// Returns a pair of set of ConstraintVariables and set of BoundsKey
+// after evaluating the expression E. Will explore E recursively, but will
+// ignore parts of it that do not contribute to the final result.
+CSetBkeyPair
+    ConstraintResolver::getExprConstraintVars(Expr *E) {
```

Comment:
Could the bounds key part of the return instead be gathered by a recursive ASTVisitor? It looks like most of the branches here just pass the bounds keys through unchanged.